### PR TITLE
 Fix #6547: logging does not work if server contains CJK characters 

### LINKED ISSUE #6547: Server log cannot be written if the server name has CJK characters
**OS:** Windows 10
**Version:** 0.1.2
**Commit/Build:** ce4899c

Server log is failed to be written if server name has non-latin characters, such as CJK.
I think if this issue is solved, then so might does #5573

- [ ] Reproducible in RCT2 (vanilla)?
- [x] Multiplayer?

**Steps to reproduce:**
1. Open a server with name as non-latin strings, such as ```서버``` (means 'server' in korean)
2. Then server log text file is failed to be written. There is no text file in /serverlog directory.

**Screenshots / Video:**
Message from the log_error() function:
image

### PR DESCRIPTION
**Pull request by telk5093**
First, I'm not good at C/C++ and github, please notice me if something is wrong.

I changed code to log serverlog correctly if the server name contains CJK characters. (#6547)
Since it was hard to deal with UTF8 file path, so I changed the path where server logs is stored like this:

 * **Before:** ``/serverlogs/{SERVERNAME}-{DATEFORMAT}.txt``
 * **After:** ``/serverlogs/{SERVERNAME}/{DATEFORMAT}.txt``

Please let me know if something is wrong.

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
Could you rebase your commits on top of `develop` rather than merging the tree back into your branch? Thank you.

**Comment by telk5093** [comment]
@AaronVanGeffen Sorry, I have no idea. How to do that?

**Comment by AaronVanGeffen** [comment]
GitHub has an [article](https://help.github.com/articles/about-git-rebase/) about it. Perhaps that helps. If not, no worries — I suppose someone can do it for you before merging.

**Comment by telk5093** [comment]
@AaronVanGeffen Dank u, it will be better to wait to do since I'm a stranger at git command.

File: src/openrct2/network/Network.cpp [hunk 1]
```diff
@@ -866,7 +866,7 @@ void Network::LoadGroups()
     group_list.at(0)->ActionsAllowed.fill(0xFF);
 }
 
-std::string Network::BeginLog(const std::string &directory, const std::string &filenameFormat)
+std::string Network::BeginLog(const std::string &directory, const std::string &midName, std::string &filenameFormat)
```
Comment (IntelOrca) [inline src/openrct2/network/Network.cpp]: You forgot to make `filenameFormat` const.

File: src/openrct2/network/Network.cpp [hunk 2]
```diff
@@ -876,7 +876,7 @@ std::string Network::BeginLog(const std::string &directory, const std::string &f
         throw std::runtime_error("strftime failed");
     }
 
-    return Path::Combine(directory, filename);
+    return Path::Combine(directory, Path::Combine(midName, filename));
```
Comment (IntelOrca) [inline src/openrct2/network/Network.cpp]: I think `Path::Combine` accepts multiple parameters.

**Comment by telk5093** [comment]
Fixed missing ``const`` and make Path::Combine() to be accepting 3 parameters. Thank you for advicing @IntelOrca

File: src/openrct2/network/Network.cpp (hunk 2 above)
Comment (rwjuk) [reply src/openrct2/network/Network.cpp]: @IntelOrca Nope, just two. https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/core/Path.cpp#L39

**Comment by Gymnasiast** [comment]
Never merge branches from us into yours. You must rebase. Or I can do it for you if you want.

**Comment by telk5093** [comment]
@Gymnasiast Sorry, I'm not good at Git. I'm glad if you do it. Thanks

**Pull request merged**